server/views/components/wip/renderField/macro.njk

```diff
@@ -0,0 +1,80 @@
+{% from "govuk/components/checkboxes/macro.njk" import govukCheckboxes %}
+{% from "govuk/components/input/macro.njk" import govukInput %}
+{% from "govuk/components/radios/macro.njk" import govukRadios %}
+{% from "moj/components/date-picker/macro.njk" import mojDatePicker %}
+
+{% macro renderField(field) %}
+  {% if field.component === "govukCheckboxes" or field.component === "govukRadios" %}
+    {# Build `items` field #}
+    {% set items = [] %}
+      {% for item in field.items %}
+          {# Build `conditional` field #}
+          {%- if item.dateRequired or item.commentRequired -%}
+            {%- set conditionalHtml -%}
+              {{ conditionalHtml(item) }}
+            {%- endset -%}
+            {%- set conditional = { html: conditionalHtml} -%}
+          {%- else -%}
+            {%- set conditional = null -%}
+          {%- endif -%}
+
+          {% set items = (items.push({
+            text: item.text,
+            value: item.value,
+            conditional: conditional
+          }), items) %}
+        {% endfor %}
+
+      {% if field.component === "govukCheckboxes" %}
+        {% set govukMacro = govukCheckboxes %}
+      {% else %}
+        {% set govukMacro = govukRadios %}
+      {% endif %}
+      {{ govukMacro({
+          name: field.name,
+          fieldset: {
+            legend: {
+              text: field.text,
+              classes: 'govuk-fieldset__legend--m'
+            }
+          },
+          items: items
+      }) }}
+  {% else %}
+    <strong>WARNING: Unknown field's component {{ field.component }}
+  {% endif %}
+{% endmacro %}
+
+{% macro conditionalHtml(item) %}
```

Comment:
this macro uses variables (`field`, `formValues`, `conditionalHtml`) internally that haven't been passed to it so i am pretty certain it will throw an exception.

Reply:
as mentioned offline...nunjucks being nunjucks it just swallows the issue and silently returns `undefined`

Reply:
https://github.com/ministryofjustice/hmpps-incident-reporting/pull/223/commits/78498059f118346f543d812d1515d3ac83670acf